python_src/src/lamp_py/ingestion/pipeline.py

```diff
@@ -71,23 +83,67 @@ def ingest(metadata_queue: Queue) -> None:
         file_prefix=DEFAULT_S3_PREFIX,
     )
 
-    ingest_files(files, metadata_queue)
+    processed_files = ingest_files(files, metadata_queue)
+
+    process_logger.log_complete()
+
+    return processed_files
+
+
+def wait_for_deletion(filenames: List[str]) -> None:
+    """
+    wait for all of the files in the filenames list to be removed from the
+    incoming bucket. log how many files are still
+    """
+    process_logger = ProcessLogger(
+        "wait_for_deletion", file_count=len(filenames)
+    )
+    process_logger.log_start()
+
+    files_to_delete = set(filenames)
+    attempts = 0
+    while len(files_to_delete) > 0 and attempts < 5:
+        existing_files = set(
+            file_list_from_s3(
+                bucket_name=os.environ["INCOMING_BUCKET"],
+                file_prefix=DEFAULT_S3_PREFIX,
+            )
+        )
+
+        files_to_delete &= existing_files
+        if len(files_to_delete) == 0:
+            break
+
+        # increment attempts and wait at most 30 seconds to check again
+        attempts += 1
+        time.sleep(min(30, int(len(files_to_delete) / 100)))
 
+    process_logger.add_metadata(
+        attempts=attempts, files_not_deleted=list(files_to_delete)
+    )
     process_logger.log_complete()
 
 
 def main() -> None:
-    """every second run jobs that are currently pending"""
+    """
+    run the ingestion pipeline
+
+    * setup metadata queue metadata writer proccess
+    * on a loop
+        * check to see if the pipeline should be terminated
+        * ingest files from incoming s3 bucket
+        * ensure processed files have been moved
+    """
     # start rds writer process
     # this will create only one rds engine while app is running
     metadata_queue, rds_process = start_rds_writer_process()
 
-    schedule.every(5).minutes.do(ingest, metadata_queue=metadata_queue)
-
+    # run the event loop every five minutes
     while True:
         check_for_sigterm(metadata_queue, rds_process)
-        schedule.run_pending()
-        time.sleep(1)
+        processed_files = ingest(metadata_queue=metadata_queue)
+        wait_for_deletion(processed_files)
+        time.sleep(60 * 5)
```

Comment:
probably need to add another `check_for_sigterm` call in here before the sleep, with this new scheduling